src/Features/LanguageServer/Protocol/RoslynLanguageServer.cs

```diff
@@ -100,10 +102,35 @@ void AddBaseServiceFromFunc<T>(Func<ILspServices, object> creatorFunc)
             }
         }
 
-        public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        public async Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
         {
             OnInitialized();
-            return Task.CompletedTask;
+
+            var capabilityRegistrationsProvider = GetLspServices().TryGetService(typeof(ICapabilityRegistrationsProvider)) as ICapabilityRegistrationsProvider;
+            if (capabilityRegistrationsProvider is null)
+            {
+                return;
+            }
+
+            var registrations = capabilityRegistrationsProvider.GetRegistrations();
```

Comment:
Seems like this should pass `clientCapabilities` in so that servers can make decisions about what/how they support various features.

In fact, shouldn't this check that dynamic registration is supported by the client?

Reply:
Good point, it should probably do both